operator/watchers/k8s_service_sync.go

```diff
@@ -229,7 +229,7 @@ func endpointsInit(slimClient slimclientset.Interface, swgEps *lock.StoppableWai
 						if oldk8sEP.DeepEqual(newk8sEP) {
 							return
 						}
-						K8sSvcCache.UpdateEndpoints(newk8sEP, swgEps)
+						K8sSvcCache.UpdateEndpoints(oldk8sEP, newk8sEP, swgEps)
```

Comment:
Directly passing the old object will become problematic once starting to use `Resource[Endpoints]` (https://github.com/cilium/cilium/pull/23977), as no longer available as part of the event. Would it be possible to retrieve the old endpoints from the already existing `ServiceCache.endpoints[<service-name>]` map (https://github.com/cilium/cilium/blob/pr%2Fchristarazi%2Fto-services-fixups/pkg/k8s/service_cache.go#L88)?

Reply:
Yep, once https://github.com/cilium/cilium/pull/23977/ is merged you don't have the old object anymore so this approach won't work. The right approach is for `ServiceCache` to keep the previous endpoint.

Reply:
Thanks. PR should be updated now!